packages/react-router/modules/Match.js

```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import invariant from "invariant";
+
+import RouterContext from "./RouterContext";
+import Route, { getContext } from "./Route";
+
+class Match extends React.Component {
+  render() {
+    Route.displayName = "Match";
+    return (
+      <RouterContext.Consumer>
+        {context => {
+          invariant(context, "You should not use <Match> outside a <Router>");
+
+          const props = getContext(this.props, context);
+
+          if (props.match) {
+            return <Route {...this.props} />;
```

Comment:
`<Match>` is not supposed to render a `<Route>`. Rather it is a new component that has a single render prop: `children`.

Reply:
Ah, I just saw @timdorr's comment above. Sorry for the confusion. `<Match>` should not render a `<Route>`. It's a different component.